### PR TITLE
uplink: don't stop deleting segments on first error

### PR DESCRIPTION
**Pull request by mniewrzal**
What: Don't stop deleting segments on first error. With current code if we are not able to delete `s0` (e.g. when nodes are not available) then we are not continuing work on next segments and just stop.

Why: Potential fast fix for https://storjlabs.atlassian.net/browse/V3-2550

Please describe the tests:
 - Test 1: automatic test added
 
Please describe the performance impact: none

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: uplink/storage/streams/store.go [hunk 1]
```diff
@@ -437,19 +437,24 @@ func (s *streamStore) Delete(ctx context.Context, path Path, pathCipher storj.Ci
 		return err
 	}
 
+	var errlist errs.Group
 	for i := 0; i < int(numberOfSegments(&stream, &streamMeta)-1); i++ {
 		currentPath, err := createSegmentPath(ctx, int64(i), path.Bucket(), encPath)
 		if err != nil {
-			return err
+			errlist.Add(err)
```
Comment (ifraixedes) [inline uplink/storage/streams/store.go]: I took a look at `createSegmentPath` and I'm wondering if these errors should be skipped.

File: uplink/storage/streams/store.go [hunk 2]
```diff
@@ -437,19 +437,24 @@ func (s *streamStore) Delete(ctx context.Context, path Path, pathCipher storj.Ci
...
+			continue
 		}
 
 		err = s.segments.Delete(ctx, currentPath)
 		if err != nil {
-			return err
+			errlist.Add(err)
```
Comment (ifraixedes) [inline uplink/storage/streams/store.go]: We are not distinguishing the errors, so the error can be returned in the metainfo or the node but I guess that's worth to carry on.

File: uplink/storage/streams/store.go [hunk 3]
```diff
@@ -437,19 +437,24 @@ func (s *streamStore) Delete(ctx context.Context, path Path, pathCipher storj.Ci
...
+			continue
 		}
 	}
 
-	return s.segments.Delete(ctx, lastSegmentPath)
+	errlist.Add(s.segments.Delete(ctx, lastSegmentPath))
+
+	return errlist.Err()
```
Comment (ifraixedes) [inline uplink/storage/streams/store.go]: We'll return errors to the client, however, the segments may be deleted despite the errors.
I'm wondering how we can inform of that to the callers.

File: uplink/storage/streams/store.go (hunk 1 above)
Comment (mniewrzal) [reply uplink/storage/streams/store.go]: this logic will be changed by #2640 so I don't think we need to care too much about it

File: uplink/storage/streams/store.go (hunk 2 above)
Comment (mniewrzal) [reply uplink/storage/streams/store.go]: we plan to reorganize deleting process so I believe then we can address this

File: uplink/storage/streams/store.go (hunk 3 above)
Comment (mniewrzal) [reply uplink/storage/streams/store.go]: yeah, this is problematic. maybe we can make different classes for errors and distinguish metainfo errors from SN errors but as I mentioned earlier we should improve the whole process in general.

**Pull request merged**